File: c/formulaDeBhasckara.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

void printValues(double *r1, double *r2)
{
    printf("R1 = %.5lf\n", *r1);
    printf("R2 = %.5lf\n", *r2);
}

double Delta(double *a, double *b, double *c)
{
    return pow(*b, 2) - 4 * (*a) * (*c);
}

double *Bhasckara(double *a, double *b, double *delta)
{
    double *x = malloc(2 * sizeof(double));
    x[0] = (-(*b) + sqrt(*delta)) / (2 * (*a));
    x[1] = (-(*b) - sqrt(*delta)) / (2 * (*a));
    return x;
}

int main()
{
    double a, b, c;
    scanf("%lf", &a);
    scanf("%lf", &b);
    scanf("%lf", &c);
    double delta = Delta(&a, &b, &c);
    if (delta > 0 && a != 0)
    {
        double *x = Bhasckara(&a, &b, &delta);
        printValues(&x[0], &x[1]);
        free(x);
    }
    else
        printf("Impossivel calcular\n");
    return 0;
}

File: apenas java/formulaDeBhasckara.rs
use std::io;

fn print_values(r1: f64, r2: f64) -> () {
    println!("R1 = {:5}", r1);
    println!("R2 = {:5}", r2);
}

fn delta(a: f64, b: f64, c: f64) -> f64 {
    (b * b) - 4.0 * a * c
}

fn bhasckara(a: f64, b: f64, delta: f64) -> (f64, f64) {
    (
        (-b + delta.sqrt()) / (2.0 * a),
        ((-b - delta.sqrt()) / (2.0 * a)),
    )
}

fn main() {
    let mut buf = String::new();
    io::stdin().read_line(&mut buf).unwrap();

    let (a, b, c): (f64, f64, f64) = buf.trim().parse().unwrap();
    // println!("{}", buf.trim().parse().unwrap());

    // let delta: f64 = delta(a, b, c);
    // if delta > 0.0 && a != 0.0 {
    //     let (r1, r2) = bhasckara(a, b, delta);
    //     print_values(r1, r2);
    // } else {
    //     println!("Impossivel calcular");
    // }
}
